tests/apidoc_login_test.go

```diff
@@ -81,6 +82,70 @@ func TestLogin(t *testing.T) {
 				},
 			})
 		})
+		// sytest: POST /login can log in as a user with just the local part of the id
+		t.Run("POST /login can log in as a user with just the local part of the id", func(t *testing.T) {
+			t.Parallel()
+			createDummyUser(t, unauthedClient, "@test_user:matrix")
+			reg := regexp.MustCompile("@([^:]*):")
+			localPart := reg.FindStringSubmatch("@test_user:matrix")
```

Comment:
Please don't do this. I know the original perl code does this but it's unclear why we need to do this rather than just hard-coding the localpart to "test_user" in this case. I would just do that. The original perl code didn't do this because it dynamically made users so you didn't know at compile time what the user localpart was.

Reply:
Since I have made the createDummyUser to return the complete userID, do you still want me to just use the localpart rather than trimming the returned userID?